src/gluonts/dataset/dataframe.py

```diff
@@ -82,17 +84,38 @@ def __post_init__(self) -> None:
         if isinstance(self.target, list) and len(self.target) == 1:
             self.target = self.target[0]
         self.one_dim_target = not isinstance(self.target, list)
-        self.process = ProcessDataEntry(
-            self.freq, one_dim_target=self.one_dim_target
-        )
+
         # store data internally as List[Tuple[str, pandas.DataFrame]]
         # if str is not empty it will be set in ``DataEntry`` as ``item_id``.
         if isinstance(self.dataframes, dict):
             self._dataframes = list(self.dataframes.items())
         elif isinstance(self.dataframes, list):
-            self._dataframes = [("", df) for df in self.dataframes]
+            self._dataframes = [self._to_tuple(df) for df in self.dataframes]
         else:  # case single dataframe
-            self._dataframes = [("", self.dataframes)]
+            self._dataframes = [self._to_tuple(self.dataframes)]
+
+        if not self.timestamp:  # infer timestamp from index
+            for i, (_, df) in enumerate(self._dataframes):
+                assert isinstance(
+                    df.index, DatetimeIndexOpsMixin
+                ), "Please set ``timestamp`` or provide a DatetimeIndex."
+                df = df.reset_index()
+                df.columns = ["_timestamp"] + df.columns.tolist()[1:]
+                self._dataframes[i] = (self._dataframes[i][0], df)
+            self.timestamp = "_timestamp"
```

Comment:
I'm thinking: wouldn't it be better to do the opposite? If `timestamp` is set, then we `.set_index(timestamp)`, and then use the index from here on; it `timestamp` is not set, then we just use the index. I feel like that way, some of the logic here would be simpler (but I didn't think it through)

Reply:
Yes, it got a bit complicated here. I'll try your suggestion tomorrow and report afterwards.